File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.java
class Solution {
    public int closestTarget(String[] words, String target, int startIndex) {
        boolean valid=false;
        for(int i=0;i<words.length;i++){
            if(words[i].equals(target)){
                valid=true;
                break;
            }
        }
        if(!valid) return -1;
        int s=startIndex, lt=0;
        int n=words.length;
        while(!words[s].equals(target)){
            s++;
            lt++;
            if(s==n)s=0;
        }
        s=startIndex;
        int rt=0;
        while(!words[s].equals(target)){
            rt++;
            s--;
            if(s==-1)s=n-1;
        }
        return Math.min(lt,rt);
    }
}

File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        if target not in words:
            return -1
        lt,rt=0,0
        s=startIndex
        while words[s] != target:
            s+=1
            rt+=1
            if s==len(words):
                s=0
        s=startIndex
        while words[s] != target:
            s-=1
            lt+=1
            if s==-1:
                s=len(words)-1
        return min(lt,rt)